flask_resources/parsers/parsers.py

```diff
@@ -6,15 +6,81 @@
 # under the terms of the MIT License; see LICENSE file for more details.
 
 """Library for easily implementing REST APIs."""
+from functools import wraps
 
 from flask import request
 from marshmallow.validate import Range, Regexp
 from webargs.fields import Int, String
-from webargs.flaskparser import parser
+from webargs.flaskparser import FlaskParser
 
+from ..context import resource_requestctx
 from .paginate import build_pagination
 
 
+class ResourceFlaskParser(FlaskParser):
+    """Customized FlaskParser for our needs."""
+
+    DEFAULT_VALIDATION_STATUS = 400
+
+
+def select_args_parser(methodview, config_parser_or_parsers):
+    """Returns ArgsParser corresponding to situation."""
+    if isinstance(config_parser_or_parsers, ArgsParser):
+        return config_parser_or_parsers
+
+    # TODO: Remove this check and rely on config validation to know that the only
+    #       other possibility is a dict
+    if not isinstance(config_parser_or_parsers, dict):
+        return
+
+    return config_parser_or_parsers.get(
+        methodview.resource_method, ArgsParser()  # default "parse nothing" parser
```

Comment:
moderate: since we only use the resource method, what about passing only that? It is more self contain and does not depend on a whole methodview (i.e. might be compatible with other view classes if only depending on a *key* - resource method in this -)

Reply:
Yes! Much better